components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -359,22 +359,20 @@ public static String formatDateTime(Date date) {
     }
 
     /**
-     * This is used to get tenant domain of thread local service provider.
+     * This is used to get tenant domain.
      *
-     * @return Service provider's tenant domain.
+     * @return user's tenant domain.
      */
-    private static String getTenantDomainFromSP() {
+    private static String getTenantDomain() {
 
         String tenantDomain;
-        ThreadLocalProvisioningServiceProvider threadLocalSP = IdentityApplicationManagementUtil
-                .getThreadLocalProvisioningServiceProvider();
-        if (threadLocalSP != null) {
-            return threadLocalSP.getTenantDomain();
-        } else if (IdentityTenantUtil.getTenantDomainFromContext() != null) {
-            return IdentityTenantUtil.getTenantDomainFromContext();
-        } else if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain() != null) {
-            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
         } else {
+            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        }
+
+        if (tenantDomain.isEmpty()){
```

Comment:
shall we use StringUtils.isBlank here

Reply:
Addressed with commit bb4f9622b8914f2c96b3c52ed8bb0b5373f6da3a